src/Promise.php

```diff
@@ -36,10 +38,11 @@ interface Promise
      * If you do not care about one of the cases, you can set the corresponding callable to null
      * The callback will be called when the value arrived and never more than once.
      *
-     * @param callable|null $onFulfilled called when a response will be available
-     * @param callable|null $onRejected  called when an exception occurs
+     * @param callable(T): V|null $onFulfilled called when a response will be available
+     * @param callable(\Exception): V|null $onRejected  called when an exception occurs
```

Comment:
doesn't `V` also need to be declared in the class annotations?

Reply:
In that case, `V` is determined by whatever `then` returns, which is possibly not related to the promise's original `T`, so there's an additional `@template V` on the method comment.  
In TypeScript, it would be:
```ts
interface Promise<T> {
    then<V>(unfulfilled: (value: T) => V | Promise<V>): Promise<V>;
}
```